xarray/backends/h5netcdf_.py

```diff
@@ -324,6 +324,13 @@ def prepare_variable(
                 raise ValueError("'zlib' and 'compression' encodings mismatch")
             encoding.setdefault("compression", "gzip")
 
+        # "zlib" is allowed as an alias for "gzip" since in
+        # libnetcdf4 allows "zlib" to be set "gzip" compression
+        # making it difficult to write code that enables common
+        # compression options for both
+        if encoding.get("compression") == "zlib":
```

Comment:
What happens if `encoding={"zlib":True, "compression": "zlib"}? This would raise an error above.

Reply:
Yes, but only in the case when the encoding is provided via `encoding`-kwarg of `to_netcdf`. Otherwise (if the encoding from the variable is used) `check_encoding` is `False` and the ValueError is not raised.

I'm assuming in normal operation `encoding` should not contain mixed styles, so the check is skipped and only run when a new encoding is provided.

Reply:
@hmaarrfk Might be good to leave a note that `h5netcdf` >= 1.5 accepts `compression="zlib"` as alias for `gzip`. Just to remind us, that we can revisit here when minimal version of `h5netcdf` is 1.5.

Reply:
> Yes, but only in the case when the encoding is provided via `encoding`-kwarg of `to_netcdf`. Otherwise (if the encoding from the variable is used) `check_encoding` is `False` and the ValueError is not raised.
> 
> I'm assuming in normal operation `encoding` should not contain mixed styles, so the check is skipped and only run when a new encoding is provided.

Then, we would need to add `zlib` to that particular check, @hmaarrf.